tests/components/color_extractor/test_service.py

```diff
@@ -0,0 +1,315 @@
+    assert state
+
+    # Validate starting values
+    assert state.state == STATE_ON
+    assert state.attributes.get(ATTR_BRIGHTNESS) == 180
+    assert state.attributes.get(ATTR_RGB_COLOR) == (255, 63, 111)
+
+    await hass.services.async_call(
+        LIGHT_DOMAIN,
+        LIGHT_SERVICE_TURN_OFF,
+        {ATTR_ENTITY_ID: LIGHT_ENTITY},
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+
+    state = hass.states.get(LIGHT_ENTITY)
+
+    assert state
+    assert state.state == STATE_OFF
+
+
+async def test_missing_url_and_path(hass):
+    """Test that nothing happens when url and path are missing."""
+    # Load our color_extractor component
+    await async_setup_component(
+        hass,
+        DOMAIN,
+        {},
+    )
+    await hass.async_block_till_done()
+
+    # Validate pre service call
+    state = hass.states.get(LIGHT_ENTITY)
+    assert state
+    assert state.state == STATE_OFF
+
+    # Missing url and path attributes, should cause error log
+    service_data = {
+        ATTR_ENTITY_ID: LIGHT_ENTITY,
+    }
+
+    await hass.services.async_call(DOMAIN, SERVICE_TURN_ON, service_data, blocking=True)
+    await hass.async_block_till_done()
+
+    # check light is still off, unchanged due to bad parameters on service call
+    state = hass.states.get(LIGHT_ENTITY)
+    assert state
+    assert state.state == STATE_OFF
+
+
+async def _async_load_color_extractor_url(hass, service_data):
+    # Load our color_extractor component
+    await async_setup_component(
+        hass,
+        DOMAIN,
+        {},
+    )
+    await hass.async_block_till_done()
+
+    # Validate pre service call
+    state = hass.states.get(LIGHT_ENTITY)
+    assert state
+    assert state.state == STATE_OFF
+
+    # Call the shared service, our above mock should return the base64 decoded fixture 1x1 pixel
+    assert await hass.services.async_call(
+        DOMAIN, SERVICE_TURN_ON, service_data, blocking=True
+    )
+
+    await hass.async_block_till_done()
+
+
+async def test_url_success(hass, aioclient_mock):
+    """Test that a successful image GET translate to light RGB."""
+    service_data = {
+        ATTR_URL: "http://example.com/images/logo.png",
+        ATTR_ENTITY_ID: LIGHT_ENTITY,
+        # Standard light service data which we pass
+        ATTR_BRIGHTNESS_PCT: 50,
+    }
+
+    # Mock the HTTP Response with a base64 encoded 1x1 pixel
+    aioclient_mock.get(
+        url=service_data[ATTR_URL],
+        content=base64.b64decode(load_fixture("color_extractor_url.txt")),
+    )
+
+    # Allow access to this URL using the proper mechanism
+    hass.config.allowlist_external_urls.add("http://example.com/images/")
+
+    await _async_load_color_extractor_url(hass, service_data)
+
+    state = hass.states.get(LIGHT_ENTITY)
+    assert state
+
+    # Ensure we turned it on
+    assert state.state == STATE_ON
+
+    # Brightness has changed, optional service call field
+    assert state.attributes[ATTR_BRIGHTNESS] == 128
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)